src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -407,6 +408,62 @@ uct_cuda_ipc_mem_dereg(uct_md_h md, const uct_md_mem_dereg_params_t *params)
     return UCS_OK;
 }
 
+static int
+uct_cuda_ipc_md_init_fabric_info(uct_cuda_ipc_md_t *md,
+                                 ucs_ternary_auto_value_t mnnvl_enable)
+{
+    int mnnvl_supported = 0;
+#if HAVE_NVML_FABRIC_INFO
+    nvmlDevice_t device;
+    ucs_status_t status;
+    char buf[64];
+
+    if (!uct_cuda_base_is_coherent() || (mnnvl_enable == UCS_NO)) {
+        goto out;
+    }
+
+    status = UCT_NVML_FUNC(nvmlInit_v2(), UCS_LOG_LEVEL_DIAG);
+    if (status != UCS_OK) {
+        goto out;
+    }
+
+    status = UCT_NVML_FUNC_LOG_ERR(nvmlDeviceGetHandleByIndex(0, &device));
+    if (status != UCS_OK) {
+        goto out_sd;
+    }
+
+    md->fabric_info.version = nvmlGpuFabricInfo_v2;
+    status                  = UCT_NVML_FUNC_LOG_ERR(
+                         nvmlDeviceGetGpuFabricInfoV(device, &md->fabric_info));
+    if (status != UCS_OK) {
+        goto out_sd;
+    }
+
+    ucs_debug("fabric_info: healthmask=%u state=%u status=%u clique=%u uuid=%s",
+              md->fabric_info.healthMask, md->fabric_info.state,
+              md->fabric_info.status, md->fabric_info.cliqueId,
+              ucs_str_dump_hex(
+                  md->fabric_info.clusterUuid, NVML_GPU_FABRIC_UUID_LEN, buf,
+                  sizeof(buf), SIZE_MAX));
+
+    if ((md->fabric_info.state != NVML_GPU_FABRIC_STATE_COMPLETED) ||
+        (md->fabric_info.status != NVML_SUCCESS)) {
+        goto out_sd;
+    }
+
+    mnnvl_supported = 1;
+
+out_sd:
+    UCT_NVML_FUNC_LOG_ERR(nvmlShutdown());
+out:
+#endif
+    if ((mnnvl_enable == UCS_YES) && !mnnvl_supported) {
+        ucs_error("multi-node NVLINK support is requested but not supported");
```

Comment:
minor: adapt message to show when it's build that does not support it?

Reply:
do we still need to have mnnvl_enable config param?

Reply:
so value of YES should override the detection logic and not generate an error

Reply:
what does override the detection logic mean? Error is printed if MNNVL support is requested but can't be initialized